internal/engine/netx/errorx/errorx.go

```diff
@@ -244,50 +280,83 @@ func toFailureString(err error) string {
 	if strings.HasSuffix(s, "TLS handshake timeout") {
 		return FailureGenericTimeoutError
 	}
-	if strings.HasSuffix(s, "no such host") {
+	if strings.HasSuffix(err.Error(), "no such host") {
 		// This is dns_lookup_error in MK but such error is used as a
 		// generic "hey, the lookup failed" error. Instead, this error
 		// that we return here is significantly more specific.
 		return FailureDNSNXDOMAINError
 	}
+	formatted := fmt.Sprintf("unknown_failure: %s", s)
+	return Scrub(formatted) // scrub IP addresses in the error
+}
 
-	// TODO(kelmenhorst): see whether it is possible to match errors
-	// from qtls rather than strings for TLS errors below.
-	//
-	// TODO(kelmenhorst): make sure we have tests for all errors. Also,
-	// how to ensure we are robust to changes in other libs?
-	//
-	// special QUIC errors
-	matched, err := regexp.MatchString(`.*x509: certificate is valid for.*not.*`, s)
-	if matched {
-		return FailureSSLInvalidHostname
+func ClassifyQUICFailure(err error) string {
```

Comment:
you mean other than `errorx.TestClassifyQUICFailure` ?

Reply:
Yes, I am wondering whether there are ways to provoke these errors in a real scenario and make sure that the QUIC library indeed returns such errors so we are actually doing things correctly beyond unit testing.

Reply:
Yes, I see, I think this should be doable for at least a few errors. Where should those integration tests live?

Reply:
> Where should those integration tests live?

They cannot live in `errorx` unless we use external testing because of circular dependencies, so probably the most rational place where to put them is inside of the `quicdialer`, `dialer`, `tlsdialer`, etc. packages.

Reply:
Yes. I did such a test in `TestErrorWrapperInvalidCertificate` (`quicdialer/errorwrapper_test.go`) but it would be good to add more.

Reply:
yes!